### PR TITLE
Integration-Test module + DynamoDB integ tests.

### PR DESCRIPTION
**Pull request by shurvitz**
*Description of changes:*
Added the Integration-Test module + integration tests for the DynamoDB connector.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: athena-dynamodb/pom.xml [hunk 1]
```diff
@@ -80,6 +80,19 @@
             <version>${log4j2Version}</version>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>athena-federation-integ-test</artifactId>
+            <version>1.0</version>
```
Comment (avirtuos) [inline athena-dynamodb/pom.xml:86]: all our modules should use the same convention for version year.week.num

File: athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Integration-tests for the DynamoDB connector using the Integration-test Suite.
+ */
+public class DynamoDBIT extends IntegrationTestBase {
```
Comment (avirtuos) [inline athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java:45]: Please rename to be consistent with our class naming convention, e.g. DynamoIntegrationTest

File: athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    {
+        ddbTableUtils.setupTableStack(stack);
+    }
+
+    /**
+     * Insert rows into the newly created DDB table.
+     */
+    protected void setupData()
+    {
+        logger.info("Setting up DB table.");
```
Comment (avirtuos) [inline athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java:101]: Do we log the table name anywhere?

File: athena-dynamodb/pom.xml [hunk 4]
```diff
@@ -80,6 +80,19 @@
...
+            <scope>test</scope>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/software.amazon.awscdk/dynamodb -->
+        <dependency>
+            <groupId>software.amazon.awscdk</groupId>
+            <artifactId>dynamodb</artifactId>
+            <version>${aws-cdk.version}</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (avirtuos) [inline athena-dynamodb/pom.xml:95]: Why is this needed? isn't it inherited from the athena-federation-integ-test module?

File: athena-federation-integ-test/README.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+### Writing Integration Tests
+
+Import the `@Test` annotation from `org.testng.annotations` and extend the Integration-Test
+class (`IntegrationTestBase`):
+
+```java
+import org.testng.annotations.Test;
+
+public class MyConnectorIT extends IntegrationTestBase
```
Comment (avirtuos) [inline athena-federation-integ-test/README.md:52]: Naming convention should suggest ItegTest instead of IT

File: athena-federation-integ-test/README.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+{
+    @Test
+    public void exampleIT()
```
Comment (avirtuos) [inline athena-federation-integ-test/README.md:55]: naming convention, not IT. we avoid consecutive capital letters in names.

File: athena-federation-integ-test/pom.xml [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+            <artifactId>aws-athena-federation-sdk</artifactId>
+            <version>2020.46.1</version>
+            <exclusions>
+                <!-- replaced with jcl-over-slf4j -->
+                <exclusion>
+                    <groupId>commons-logging</groupId>
+                    <artifactId>commons-logging</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
```
Comment (avirtuos) [inline athena-federation-integ-test/pom.xml:33]: Why does this need the federation SDK? I thought this integration test only targets Athena?

File: athena-dynamodb/pom.xml [hunk 8]
```diff
@@ -80,6 +80,19 @@
             <version>${log4j2Version}</version>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>athena-federation-integ-test</artifactId>
+            <version>1.0</version>
+            <scope>test</scope>
```
Comment (avirtuos) [inline athena-dynamodb/pom.xml:87]: can you confirm that this module does _not_ end up in the final lambda jar? you can do this by unziping the final DDB jar and ensuring the classes from this module don't show up in it. Also confirm no CDK dependencies show up. This is important because jar size is a tough limit in lambda.

**Review by avirtuos — CHANGES_REQUESTED** [review]
Looks pretty good. Only minior changes requested.

File: athena-federation-integ-test/pom.xml (hunk 7 above)
Comment (shurvitz) [reply athena-federation-integ-test/pom.xml:33]: Removed Federation SDK dependency and replaced with Athena and Guava dependencies.

File: athena-dynamodb/pom.xml (hunk 1 above)
Comment (shurvitz) [reply athena-dynamodb/pom.xml:86]: Replaced version 1.0 with 2020.46.1

File: athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java (hunk 3 above)
Comment (shurvitz) [reply athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java:101]: It's there when we log the stack template, but I added it to this log message as well.

File: athena-dynamodb/pom.xml (hunk 4 above)
Comment (shurvitz) [reply athena-dynamodb/pom.xml:95]: This one is not inherited. The `dynamodb` artifact is specific to the DynamoDB connector so it only exists in this pom.

File: athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java (hunk 2 above)
Comment (shurvitz) [reply athena-dynamodb/src/test/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBIT.java:45]: Changed to `DynamoDBIntegrationTest` (consistent with other test files, e.g: `DynamoDBRecordHandlerTest`

File: athena-federation-integ-test/README.md (hunk 5 above)
Comment (shurvitz) [reply athena-federation-integ-test/README.md:52]: Changed to `MyConnectorIntegrationTest`.

File: athena-federation-integ-test/README.md (hunk 6 above)
Comment (shurvitz) [reply athena-federation-integ-test/README.md:55]: Changed to `exampleIntegrationTest`.

File: athena-dynamodb/pom.xml (hunk 8 above)
Comment (shurvitz) [reply athena-dynamodb/pom.xml]: confirmed.

File: athena-dynamodb/pom.xml (hunk 4 above)
Comment (avirtuos) [reply athena-dynamodb/pom.xml:95]: ahhh, this is the DDB module of CDk. got it.

**Review by avirtuos — APPROVED** [review]
